Op that takes handle scope and a cppgc argument doesn't compile with `#[op2(fast)]`

Repro

```
pub struct Foo;
impl deno_core::GarbageCollected for Foo {}
#[op2(fast)]
pub fn op_repro(_scope: &mut v8::HandleScope, #[cppgc] _foo: &Foo) {}
```

Fails to compile with

```
error[E0502]: cannot borrow `*fast_api_callback_options.isolate` as mutable because it is also borrowed as immutable
   --> repro/main.rs:3:1
    |
3 | #[op2(fast)]
    | ^^^^^^^^^^^-
    | |          |
    | |          immutable borrow might be used here, when `scope` is dropped and runs the `Drop` code for type `CallbackScope`
    | mutable borrow occurs here
    | immutable borrow occurs here
    |
    = note: this error originates in the attribute macro `op2` (in Nightly builds, run with -Z macro-backtrace for more info)
```

If you remove the fast attr, then `op2` errors with `This op is fast-compatible and should be marked as (fast)`.

The only way to get it to compile is with the `nofast` attr.

--- a/ops/op2/dispatch_fast.rs
+++ b/ops/op2/dispatch_fast.rs
@@ -102,24 +102,35 @@ impl FastSignature {
     for arg in &self.args {
       match arg {
         FastArg::Actual { arg, name_out, .. }
-        | FastArg::Virtual { name_out, arg } => call_args.push(
-          map_v8_fastcall_arg_to_arg(generator_state, name_out, arg).map_err(
-            |s| V8SignatureMappingError::NoArgMapping(s, arg.clone()),
-          )?,
-        ),
+        | FastArg::Virtual { name_out, arg } => {
+          if !matches!(arg, Arg::Ref(_, Special::HandleScope)) {
+            call_args.push(
+              map_v8_fastcall_arg_to_arg(generator_state, name_out, arg)
+                .map_err(|s| {
+                  V8SignatureMappingError::NoArgMapping(s, arg.clone())
+                })?,
+            )
+          } else {
+            generator_state.needs_scope = true;
+          }
+        }
         FastArg::CallbackOptions | FastArg::PromiseId => {}
       }
     }
     Ok(call_args)
   }
 
-  pub(crate) fn call_names(&self) -> Vec<Ident> {
+  pub(crate) fn call_names(&self) -> Vec<TokenStream> {
     let mut call_names = vec![];
     for arg in &self.args {
       match arg {
-        FastArg::Actual { name_out, .. }
-        | FastArg::Virtual { name_out, .. } => {
-          call_names.push(name_out.clone())
+        FastArg::Actual { name_out, arg, .. }
+        | FastArg::Virtual { name_out, arg } => {
+          if matches!(arg, Arg::Ref(_, Special::HandleScope)) {
+            call_names.push(quote!(&mut scope));
+          } else {
+            call_names.push(quote!(#name_out));
+          }
         }
         FastArg::CallbackOptions | FastArg::PromiseId => {}
       }
@@ -426,7 +437,6 @@ pub(crate) fn generate_dispatch_fast(
   let with_stack_trace = if generator_state.needs_stack_trace {
     generator_state.needs_opctx = true;
     generator_state.needs_scope = true;
-    generator_state.needs_fast_scope = false;
 
     gs_quote!(generator_state(opctx, scope, opstate) =>
     (if #opctx.enable_stack_trace {
@@ -489,7 +499,7 @@ pub(crate) fn generate_dispatch_fast(
   };
 
   let with_isolate = if generator_state.needs_fast_isolate
-    && !generator_state.needs_fast_scope
+    && !generator_state.needs_scope
     && !generator_state.needs_stack_trace
   {
     generator_state.needs_fast_api_callback_options = true;
@@ -634,7 +644,6 @@ fn map_v8_fastcall_arg_to_arg(
     opctx,
     js_runtime_state,
     scope,
-    needs_scope,
     needs_opctx,
     needs_fast_api_callback_options,
     needs_fast_isolate,
@@ -722,8 +731,7 @@ fn map_v8_fastcall_arg_to_arg(
       quote!(let #arg_ident = &mut ::std::cell::RefCell::borrow_mut(&#opctx.state);)
     }
     Arg::Ref(_, Special::HandleScope) => {
-      *needs_scope = true;
-      quote!(let #arg_ident = &mut #scope;)
+      unreachable!()
     }
     Arg::RcRefCell(Special::OpState) => {
       *needs_opctx = true;

--- a/ops/op2/generator_state.rs
+++ b/ops/op2/generator_state.rs
@@ -45,7 +45,6 @@ pub struct GeneratorState {
   pub needs_args: bool,
   pub needs_retval: bool,
   pub needs_scope: bool,
-  pub needs_fast_scope: bool,
   pub needs_fast_isolate: bool,
   pub needs_isolate: bool,
   pub needs_opstate: bool,

--- a/ops/op2/mod.rs
+++ b/ops/op2/mod.rs
@@ -200,7 +200,6 @@ pub(crate) fn generate_op2(
     needs_retval: false,
     needs_scope: false,
     needs_fast_isolate: false,
-    needs_fast_scope: false,
     needs_isolate: false,
     needs_opctx: false,
     needs_opstate: false,

--- a/ops/op2/test_cases/sync/result_scope.out
+++ b/ops/op2/test_cases/sync/result_scope.out
@@ -99,10 +99,7 @@ pub const fn op_void_with_result() -> ::deno_core::_ops::OpDecl {
                 >::cast_unchecked(unsafe { fast_api_callback_options.data })
                     .value() as *const deno_core::_ops::OpCtx)
             };
-            let result = {
-                let arg0 = &mut scope;
-                Self::call(arg0)
-            };
+            let result = { Self::call(&mut scope) };
             let result = match result {
                 Ok(result) => result,
                 Err(err) => {

--- a/testing/checkin/runner/ops.rs
+++ b/testing/checkin/runner/ops.rs
@@ -94,6 +94,9 @@ impl TestObjectWrap {
   }
 
   #[fast]
+  fn with_scope_fast(&self, _scope: &mut v8::HandleScope) {}
+
+  #[fast]
   #[rename("with_RENAME")]
   fn with_rename(&self) {}
 
@@ -289,3 +292,10 @@ impl TestEnumWrap {
 pub fn op_nop_generic<T: SomeType + 'static>(state: &mut OpState) {
   state.take::<T>();
 }
+
+pub struct Foo;
+
+impl deno_core::GarbageCollected for Foo {}
+
+#[op2(fast)]
+pub fn op_scope_cppgc(_scope: &mut v8::HandleScope, #[cppgc] _foo: &Foo) {}

